anvil/src/config.rs

```diff
@@ -326,93 +499,19 @@ impl NodeConfig {
 
     /// Prints the config info
     pub fn print(&self, fork: Option<&ClientFork>) {
+        if self.config_out.is_some() {
+            let config_out = self.config_out.as_deref().unwrap();
+            let f =
+                File::create(config_out).expect("Unable to create anvil config description file");
+            let mut f = BufWriter::new(f);
```

Comment:
I think you probably can instead avoid the "serde_json::to string" in as_json, and return the serde_json::Value type and here do serde_json::to_writer. Basically same thing but without you needing to manage the file descriptor and do write_all.